walletconnectv2/src/main/kotlin/com/walletconnect/walletconnectv2/engine/model/mapper/EngineMapper.kt

```
@@ -77,34 +72,26 @@ internal fun SessionParamsVO.EventParams.toEngineDOSessionEvent(topic: TopicVO):
 @JvmSynthetic
 internal fun SessionVO.toEngineDOApprovedSessionVO(topic: TopicVO): EngineDO.Session =
     EngineDO.Session(
-        topic,
-        expiry,
-        accounts,
+        topic, expiry, accounts, methods, events,
         EngineDO.AppMetaData(
             selfMetaData?.name ?: String.Empty,
             selfMetaData?.description ?: String.Empty,
             selfMetaData?.url ?: String.Empty,
-            selfMetaData?.icons?.map { iconUri -> iconUri } ?: listOf()),
-        EngineDO.SessionPermissions(EngineDO.SessionPermissions.JsonRpc(methods), getEvents(events)),
-        EngineDO.Blockchain(chains),
+            selfMetaData?.icons?.map { iconUri -> iconUri } ?: listOf())
     )
 
 @JvmSynthetic
 internal fun SessionVO.toEngineDOApprovedSessionVO(): EngineDO.Session =
     EngineDO.Session(
-        topic, expiry,
-        accounts, selfMetaData?.toEngineDOAppMetaData(),
-        EngineDO.SessionPermissions(EngineDO.SessionPermissions.JsonRpc(methods), getEvents(events)),
-        EngineDO.Blockchain(chains),
+        topic, expiry, accounts, methods, events,
+        selfMetaData!!.toEngineDOAppMetaData()
```

using the `!!` can be dangerous. We should probably just throw an exception if metadata is null

---

Ya true. However, the selfMetaData should not be nullable. I need to review this again.